[request] libiconv/1.18

### Package Name/Version

libiconv/1.18

### Changelog

https://ftp.gnu.org/pub/gnu/libiconv/libiconv-1.18.tar.gz

### Context about the new update

This version fixes a build error on GCC 15, as it now defaults to C23, changing what a function declaration with no arguments means:
```
${CONAN_HOME}/p/b/libic264b9d4d8d94a/b/src/lib/loop_wchar.h:39:17: error: conflicting types for 'mbrtowc'; have 'size_t(void)' {aka 'long unsigned int(void)'}
   39 |   extern size_t mbrtowc ();
      |                 ^~~~~~~
${CONAN_HOME}/p/b/libic264b9d4d8d94a/b/src/lib/loop_wchar.h:356:15: error: too many arguments to function 'mbrtowc'; expected 0, have 4
  356 |         res = mbrtowc(&wc,buf,bufcount,&state);
      |               ^~~~~~~ ~~~
```

--- a/recipes/libiconv/all/conandata.yml
+++ b/recipes/libiconv/all/conandata.yml
@@ -1,4 +1,7 @@
 sources:
+  "1.18":
+    url: "https://ftp.gnu.org/gnu/libiconv/libiconv-1.18.tar.gz"
+    sha256: "3b08f5f4f9b4eb82f151a7040bfd6fe6c6fb922efe4b1659c66ea933276965e8"
   "1.17":
     url: "https://ftp.gnu.org/gnu/libiconv/libiconv-1.17.tar.gz"
     sha256: "8f74213b56238c85a50a5329f77e06198771e70dd9a739779f4c02f65d971313"

--- a/recipes/libiconv/config.yml
+++ b/recipes/libiconv/config.yml
@@ -1,4 +1,6 @@
 versions:
+  "1.18":
+    folder: all
   "1.17":
     folder: all
   "1.16":

